src/Orleans.Runtime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -586,7 +586,8 @@ public async Task<AddressAndTag> RegisterAsync(ActivationAddress address, bool s
             if (hopCount > 0 && forwardAddress != null)
             {
                 await Task.Delay(RETRY_DELAY);
-                forwardAddress = this.CheckIfShouldForward(address.Grain, hopCount, "RegisterAsync(recheck)");
+                forwardAddress = this.CheckIfShouldForward(address.Grain, hopCount, "RegisterAsync");
+                this.log.LogWarning($"RegisterAsync - It seems we are not the owner of activation {address}, trying to forward it to {forwardAddress} (hopCount={hopCount})");
```

Comment:
These semi-duplicate message strings bother me a bit. Can we collapse them?

Reply:
We could, but we would end up creating multiple string anyway, since there is the operation, the grain identity, ect. And we will propably lose the variable name in the log string if we use a string.Format

Reply:
These should use structured logging rather than string templating, imo

Reply:
This code will be replaced soon anyway, let's keep it simple for now